asteroid/models/demask.py

```diff
@@ -0,0 +1,154 @@
+                n_feats_input = (self.encoder.filterbank.n_filters) // 2
+        elif self.input_type == "cat":
+            if fb_type == "stft":
+                n_feats_input = (
+                    (self.encoder.filterbank.n_filters // 2) + 1 + self.encoder.filterbank.n_filters
+                )
+            else:
+                n_feats_input = (
+                    self.encoder.filterbank.n_filters // 2
+                ) + self.encoder.filterbank.n_filters
+        elif self.input_type == "reim":
+            n_feats_input = self.encoder.filterbank.n_filters
+        else:
+            print("Input type should be either mag, reim or cat")
+            raise NotImplementedError
+
+        if self.output_type == "mag":
+            if fb_type == "stft":
+                n_feats_output = self.encoder.filterbank.n_filters // 2 + 1
+            else:
+                n_feats_output = self.encoder.filterbank.n_filters // 2
+        elif self.input_type == "reim":
+            n_feats_output = self.encoder.filterbank.n_filters
+        else:
+            print("Input type should be either mag or reim")
+            raise NotImplementedError
+
+        net = [norms.get(norm_type)(n_feats_input)]
+        in_chan = n_feats_input
+        for layer in range(len(hidden_dims)):
+            net.extend(
+                [
+                    nn.Conv1d(in_chan, hidden_dims[layer], 1),
+                    norms.get(norm_type)(hidden_dims[layer]),
+                    activations.get(activation)(),
+                    nn.Dropout(dropout),
+                ]
+            )
+            in_chan = hidden_dims[layer]
+
+        net.extend([nn.Conv1d(in_chan, n_feats_output, 1), activations.get(mask_act)()])
+
+        self.masker = nn.Sequential(*net)
+
+    def forward(self, wav):
+
+        # Handle 1D, 2D or n-D inputs
+        was_one_d = False
+        if wav.ndim == 1:
+            was_one_d = True
+            wav = wav.unsqueeze(0).unsqueeze(1)
+        if wav.ndim == 2:
+            wav = wav.unsqueeze(1)
+        # Real forward
+        tf_rep = self.encoder(wav)
+        if self.input_type == "mag":
+            est_masks = self.masker(take_mag(tf_rep))
+        elif self.input_type == "reim":
+            est_masks = self.masker(tf_rep)
+        elif self.input_type == "cat":
+            est_masks = self.masker(take_cat(tf_rep))
+        else:
+            raise NotImplementedError
+
+        if self.output_type == "mag":
+            masked_tf_rep = est_masks.repeat(1, 2, 1) * tf_rep
+        elif self.output_type == "reim":
+            masked_tf_rep = est_masks * tf_rep.unsqueeze(1)
+        else:
+            raise NotImplementedError
+
+        out_wavs = torch_utils.pad_x_to_y(self.decoder(masked_tf_rep), wav)
+        if was_one_d:
+            return out_wavs.squeeze(0)
+        return out_wavs
+
+    def get_model_args(self):
+        """ Arguments needed to re-instantiate the model. """
+
+        model_args = self.__dict__
```

Comment:
This is `self.__dict__`
```
{'training': True, '_parameters': OrderedDict(), '_buffers': OrderedDict(), '_non_persistent_buffers_set': set(), '_backward_hooks': OrderedDict(), '_forward_hooks': OrderedDict(), '_forward_pre_hooks': OrderedDict(), '_state_dict_hooks': OrderedDict(), '_load_state_dict_pre_hooks': OrderedDict(), '_modules': OrderedDict([('encoder', Encoder(
  (filterbank): STFTFB()
)), ('decoder', Decoder(
  (filterbank): STFTFB()
)), ('masker', Sequential(
  (0): GlobLN()
  (1): Conv1d(257, 1024, kernel_size=(1,), stride=(1,))
  (2): GlobLN()
  (3): ReLU()
  (4): Dropout(p=0, inplace=False)
  (5): Conv1d(1024, 257, kernel_size=(1,), stride=(1,))
  (6): ReLU()
))]), 'input_type': 'mag', 'output_type': 'mag', 'hidden_dims': [1024], 'dropout': 0, 'activation': 'relu', 'mask_act': 'relu', 'norm_type': 'gLN', 'fb_type': 'stft', 'n_filters': 512, 'stride': 256, 'kernel_size': 512, 'fb_kwargs': {}}

```
is this really what we want ?

Reply:
forgot to filter  the underscore

Reply:
maybe this can do it `model_args = {k: v for k, v in self.__dict__.items() if not k.startswith('_') and k != "training"}`